test/customscript.spec.ts

```diff
@@ -85,110 +67,97 @@ describe('CustomScriptIdentity', () => {
-            controlBlock: Buffer.alloc(33),
-          },
-        ],
-      })
-      .toBase64();
-
-    const signed = await blindAndSignPset(
-      pset,
-      utxos,
-      [id],
-      [addr.confidentialAddress],
-      [addr.confidentialAddress],
-      true
+    expect(addr.confidentialAddress).toBeDefined();
+    expect(addr.blindingPrivateKey).toBeDefined();
+    expect(addr.derivationPath).toBeDefined();
+    expect(addr.taprootHashTree).toBeDefined();
+    expect(addr.taprootInternalKey).toBe(
+      '0000000000000000000000000000000000000000000000000000000000000000'
     );
-    const txid = await broadcastTx(signed);
-    expect(txid).toBeDefined();
+    expect(addr.publicKey).toBeDefined();
+    expect(addr.contract).toBeDefined();
   });
 
-  test('should be able to auto-sign CHECKSIG tapscript owned by marina', async () => {
-    const covenantLeaf = `$${TEST_NAMESPACE} OP_CHECKSIG`;
-    const template = `eltr(c6047f9441ed7d6d3045406e95c07cd85c778e4b8cef3ca7abac09b95c709ee5, { asm(${covenantLeaf}), asm(OP_FALSE) })`;
-    const id = makeRandomCustomScriptIdentity(template);
-    const addr = await id.getNextAddress();
-
-    await faucet(addr.confidentialAddress, 10000);
-    const utxos = await getUnspents(addr);
-    expect(utxos.length).toBeGreaterThanOrEqual(1);
-
-    const { pset } = await createSendPset(
-      [
-        {
-          address: addr.confidentialAddress,
-          asset: networks.regtest.assetHash,
-          value: 1000,
-        },
-      ],
-      utxos,
-      networks.regtest.assetHash,
-      () => addr.confidentialAddress,
-      'regtest'
-    );
+  test('should be able to instatiate a contract identity, fund the contract and spend those funds', async () => {
+    const template = JSON.stringify(transferWithCaptchaArtifact);
+    const random = makeRandomCustomScriptIdentity(template);
+    expect(random.contract.namespace).toBe(TEST_NAMESPACE);
+    expect(random.contract.template).toBeDefined();
+
+    const addr = await random.getNextAddress({ sum: 7 });
 
-    const signed = await blindAndSignPset(
-      pset,
-      utxos,
-      [id],
-      [addr.confidentialAddress],
-      [addr.confidentialAddress],
-      true
+    expect(addr.confidentialAddress).toBeDefined();
+    expect(addr.blindingPrivateKey).toBeDefined();
+    expect(addr.derivationPath).toBeDefined();
+    expect(addr.taprootHashTree).toBeDefined();
+    expect(addr.taprootInternalKey).toBe(
+      '0000000000000000000000000000000000000000000000000000000000000000'
+    );
+    expect(addr.publicKey).toBeDefined();
+    expect(addr.contract).toBeDefined();
+
+    await faucet(addr.confidentialAddress, 0.01);
+
+    const [utxo] = await getUnspents(addr);
+
+    const instance = addr.contract.from(utxo.txid, utxo.vout, utxo.prevout);
+    const signer: Signer = {
+      signTransaction: async (psetBase64: string): Promise<string> => {
+        return random.signPset(psetBase64);
+      },
+    };
```

Comment:
@tiero something's wrong with this Signer interface: the error retuned by broadcastTx suggests that the signature in the input witness is not valid. Any idea?

Reply:
need to be investigate how signPset is producing the signature, has been updated to support taproot signing (hasForWitnessv1 etc..)?

Reply:
Seems there is some taproot logic there, so it should work. 

Can you inspect how the witness is for that covenant input you trying to unlock?

ideally, the end result would be to update the `sig` made with `ecc.signSchnorr` inside the input and return that psbt

```ts
ptx.updateInput(index, {
          tapScriptSig: [
            {
              leafHash,
              pubkey: keyPair.publicKey.slice(1),
              signature: sig,
            },
          ],
        });

```

Reply:
maybe @louisinger can help, that logic in signPset is huge and poorly documented as of now (plus does a lot of magic tricks, so would be good occasion to simplify and strip it down)

Reply:
related to https://github.com/altafan/marina/pull/1